selfdrive/controls/lib/pid.py

```diff
@@ -60,27 +61,33 @@ def reset(self):
     self.saturated = False
     self.control = 0
 
-  def update(self, setpoint, measurement, speed=0.0, check_saturation=True, override=False, feedforward=0., deadzone=0., freeze_integrator=False):
+  def update(self, setpoint, measurement, last_output=None, speed=0.0, check_saturation=True, override=False, feedforward=0., deadzone=0., freeze_integrator=False):
     self.speed = speed
 
     error = float(apply_deadzone(setpoint - measurement, deadzone))
     self.p = error * self.k_p
     self.f = feedforward * self.k_f
 
-    if override:
-      self.i -= self.i_unwind_rate * float(np.sign(self.i))
+    # Clip integrator based on the last output value
+    if last_output is not None:
+      if not freeze_integrator:
+        i_bf = self.p + self.i + self.f - last_output
+        self.i = self.i +  self.k_i * (error * self.i_rate - i_bf * self.i_bf_rate)
     else:
```

Comment:
There are now entirely different codepaths to do very similar things. Can we put it all in the same codepath? Or if this just a temporary thing, add TODOs making this clear what should still be changed.